utils/tinystr/src/ascii.rs

```diff
@@ -156,6 +186,28 @@ macro_rules! check_is {
             true
         }
     };
+    ($self:ident, $check:ident, CASE, $check_u8_0:ident, $check_u8_1:ident) => {
+        if N <= 4 {
+            Aligned4::from_bytes(&$self.bytes).$check()
+        } else if N <= 8 {
+            Aligned8::from_bytes(&$self.bytes).$check()
+        } else {
+            // Won't panic because N is > 8
+            if $self.bytes[0].$check_u8_0() {
+                return false;
+            }
+            let mut i = 1;
+            // Won't panic because self.bytes has length N
+            #[allow(clippy::indexing_slicing)]
+            while i < N && $self.bytes[i] != 0 {
+                if $self.bytes[i].$check_u8_1() {
+                    return false;
+                }
+                i += 1;
+            }
+            true
+        }
+    };
 }
```

Comment:
Thanks for looking at this. The problem is that `char::is_ascii_uppercase` returns `false` for non-alpha characters. The following two statements are equivalent for us:

1. `c.is_ascii_alphabetic() && !c.is_ascii_uppercase()`
2. `c.is_ascii_lowercase()`

I don't see a way to simplify the macro since some cases need a `!` and other cases don't. I'd rather land the PR as-is, and if you have ideas on how to improve it, you can do it in a follow-up.

Reply:
I see. Could you take out the `CASE` though? I think it's weird to select 'overloads' like that, and there is no ambiguity without it

Reply:
There are two differences between the "overload" and the normal one:

1. Special case for the first character
2. Inverting the conditionals

I'll make a follow-up PR clarifying this.

Reply:
#1701